modules/cudacodec/misc/python/test/test_cudacodec.py

```diff
@@ -93,6 +91,24 @@ def test_reader(self):
             else:
                 self.skipTest(e.err)
 
+    def test_map_hsitogram(self):
+        try:
+            hist = cv.cuda_GpuMat((1,256), cv.CV_8UC1)
+            hist.setTo(1)
+            hist_host = cv.cudacodec.MapHist(hist)
+            self.assertTrue(hist_host.shape == (256, 1) and isinstance(hist_host, np.ndarray))
+        except cv.error as e:
```

Comment:
Maybe we should just catch and print the exception as the error messages are currently incorrect.  Unless I am mistaken there are five possible status codes at the moment
1) `cv.Error.StsError` - General error
    1) if the decoder failed to initialize (unsupported codec, functionality i.e. histogram output) or
    2) parsing/decoding failed.
2) `cv.Error.StsNotImplemented` - if the FFmpeg backend is no available
3) `cv.Error.StsUnsupportedFormat` - if the source cannot be opened
4) `cv.Error. GpuNotSupported` - if there is no CUDA or NVCUVID
5) `cv.Error.GPU_API_CALL_ERRROR` - if any of the CUDA functions failed which could be for many reasons, invalid arguments, out of memory etc. etc.

What do you think?

Reply:
I removed try-catch. The function should work independently from NVCUVID.

Reply:
The same is true for `test_reader`, should we just print the exception message as the current `SKipTest` messages are incorrect?

Reply:
Most probably yes. I'm trying to stabilize tests and enable them in CI. After that I want to refactor Python CUDA tests and CI should help a lot.